### PR TITLE
NFDIV-1930 : Applicant solicitor NOP

### PR DESCRIPTION
**Pull request by sunilnatikar**
### JIRA link ###
https://tools.hmcts.net/jira/browse/NFDIV-1930

### Change description ###
Changes done to generate and send applicant solicitor NOP when solicitor is registered and its not a solicitor service.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java [hunk 1]
```diff
@@ -68,6 +73,12 @@ private void generateSoleNoticeOfProceedings(CaseData caseData, Long caseId) {
         final boolean isApplicant1Represented = caseData.getApplicant1().isRepresented();
         final boolean isApplicant2Represented = caseData.getApplicant2().isRepresented();
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java:75]: Is it worth making this an if/else now?

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java [hunk 2]
```diff
@@ -68,6 +73,12 @@ private void generateSoleNoticeOfProceedings(CaseData caseData, Long caseId) {
...
 
+        if (isApplicant1Represented
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java:76]: Is it worth making this and the !isApplicant1Represented an if/else now

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java [hunk 3]
```diff
@@ -68,6 +73,12 @@ private void generateSoleNoticeOfProceedings(CaseData caseData, Long caseId) {
...
+            && caseData.getApplicant1().getSolicitor().hasOrgId()
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java:77]: is this applicable for paper cases ?

Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java:77]: this is for online only sole and joint application as confirmed with Lalita

**Pull request merged**